### PR TITLE
OOP School Library: Add Basic UI

### PR DESCRIPTION
**Pull request by DJ-MrJay**
- Created <code>app.rb</code> file to serve as console app entry-point
- Created <code>main.rb</code> to define the entry point with a method called <code>main</code> that is invoked at the end

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,15 +1,17 @@
 require_relative './nameable'
+require './rental'
 
 class Person < Nameable
-  attr_accessor :name, :age
+  attr_accessor :name, :age, :rentals
   attr_reader :id
 
-  def initialize(age, name = 'Unknown', parent_permission: true)
+  def initialize(age, name = 'unknown', parent_permission: true)
```
Comment (cynthiainga) [inline person.rb]: - [ ] Good job so far but you still have some issues that need to be fixed. When I try to create a student the console app failed `wrong number of arguments`. Please kindly fix that 🙃 🙃 

Screen Shot 2023-01-07 at 14 16 31

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    list_books
+    selected_book = gets.chomp.to_i
+    puts 'Select a person from the following list by number'
+    list_people
+    selected_person = gets.chomp.to_i
+    print 'Date (DD/MM/YYYY): '
+    date = gets.chomp
+    book = @books[selected_book]
+    person = @people[selected_person]
+    rental_item = Rental.new(date, book, person)
```
Comment (cynthiainga) [inline app.rb]: - [ ] Also when I try to create `rental` the console app failed, same error `wrong number of arguments`, kindly fix that as well 🙃 😉 

Screen Shot 2023-01-07 at 14 20 47

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @DJ-MrJay, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- [ ] No linters errors ✔️ 
- [ ] Correct Gitflow ✔️ 
- [ ] Good commit history ✔️ 

## Required Changes ♻️

- [ ] Please kindly check the comments under the review 👇 👇 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by victorokoroji — APPROVED** [review]
Hi @DJ-MrJay, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request is following the Git Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- The commit messages are meaningful ✅ 
- App is working properly ✅ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by thecodechaser — COMMENTED** [review]
## STATUS: APPROVED 🟢

## Hi @DJ-MrJay👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**